File: api/QLThuVien.Data/Models/Book.cs
using QLThuVien.Data.Models;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace QLThuVien.Business.Models;

public class Book
{
    public Guid Id { get; set; }
    [Required]
    public required string AuthorName { get; set; }
    [Required]
    public required string Title { get; set; }
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
    [Required]
    public required string PublisherName { get; set; }
    [Required]
    public required DateOnly PublishDate { get; set; }
    [DefaultValue(0)]
    public required int Count { get; set; }
    public Guid CategoryId { get; set; }
    public Category Category { get; set; }
    public IEnumerable<Borrow> Borrows { get; set; }
    public IEnumerable<Rating> Ratings { get; set; }
}

File: frontend/src/app/view-models/book.ts
export class Book {
  id: string = 'no-id';
  title: string = 'Book Title';
  authorName: string = 'Unkown Author';
  publisherName: string = 'Unkown Publisher';
  description?: string | null = 'No Description';
  categoryName: string = 'Uncategorized';
  imageUrl: string = '/images/book.png';
  publishDate: Date = new Date();
  averageRating: number = 3;
  count: number = 0;
}
